### DISCUSSION TITLE
Custom client and worldwide availability  (random devices)

### DISCUSSION THREAD

**Discussion by pdifeo** [rustdesk/rustdesk-server-pro#307] [Q&A]
After a long time, I made a test run of creating a custom client.

The premise is that I have never given the server key to anyone.

Strangely enough, I find a number of accesses (see below) that I have never had until now. The good thing is that I have disabled the new devices. The server has been active since last year and no one has ever logged in until 1 hour after the custom client was created.

Am I missing something ? I understand that custom client compilations are made available on the network . Is that so? 

For me, all the security I had built is gone. I would like to understand this better. 

![image](https://github.com/user-attachments/assets/a2d71566-c91c-412d-9f8d-dff84c2a5ed2)

**Comment by pdifeo** [comment]
Can anyone help me understand?

I can't understand how they were able to connect to my server. 

This only happened after I created a custom client. 

How is this possible? The only explanation I give myself is that the builds are accessible to everyone . Is that so?

**Comment by dinger1986** [comment]
its most likely AV doing it, mine has started doing as well. For added security make sure you enable The control end need to login before access under settings/others

**Reply by pdifeo** [reply]
Excuse my ignorance or it escapes me at the moment. What do you mean by ‘AV’?

I understand that for the future the problem is solved, but what about those builds that have already been downloaded?

is it appropriate to change key ?

**Reply by dinger1986** [reply]
Anti Virus, when it samples the executables it can do that, I have seen it before with other systems.

You could change the key or leave it as is and enable that other setting then noone can use your server anyway

**Reply by pdifeo** [reply]
Disabling the registration of new clients is restrictive. It means that there must always be an administrator to enable the new client. 

For the key change, doing this on all installed clients means accessing them with another remote access software. This has become a big problem.

**Reply by dinger1986** [reply]
it doesnt disable adding new clients, it enforces login to use rustdesk to take control of another device, so without a username and password noone can use your server

**Reply by pdifeo** [reply]
Not practicable for occasional customers

**Reply by dinger1986** [reply]
Why? They wouldn't be taking control of you. You are taking control of them

**Reply by pdifeo** [reply]
@dinger1986  we have a different view of occasional customers
1) You have to make them download the client
2) They have to run it
3) They have to log in to the server . **Why consume a licence for an occasional client?**
4) They have to give you a temporary ID and password.

That means spending at least 20 minutes on the phone explaining what they have to do.  Not all customers are computer trained

**Reply by dinger1986** [reply]
Why would they need to log in?

**Reply by Nawaki89** [reply]
> @dinger1986  we have a different view of occasional customers
> 
> 1) You have to make them download the client
> 
> 2) They have to run it
> 
> 3) They have to log in to the server . **Why consume a licence for an occasional client?**
> 
> 4) They have to give you a temporary ID and password.
> 
> 
> 
> That means spending at least 20 minutes on the phone explaining what they have to do.  Not all customers are computer trained
> 
> 

I think he means that for you to be able to remote control a client you need to login.  Not the computer you are connecting to.

**Reply by dinger1986** [reply]
@Nawaki89 yes that's what I meant, just login on the supporters machine not supportee

**Reply by pdifeo** [reply]
Ok, thanks

**Comment by FCSTechAdmin** [comment]
I am also seeing rogue, uninvited, devices showing up and I don't see an explanation for them here. Is there an explanation? 
![Screenshot From 2025-02-17 10-40-45](https://github.com/user-attachments/assets/11bbea1d-f55b-45e6-924a-855d2b5f94ca)

**Comment by OtterKring** [comment]
Same here. I set up the 30-day pro server last week. When we connected the first 2 machines, a third one from "outer space" appeared in the device list, showing the IP of our firewall. Today there are two more.

* disabling new devices is not an option. New devices would be created all the time and it would require admins to manually enable the devices after installation. This is a management overhead.
* forcing admins to logon is ok (Azure SSO already set up), they will be set up in the portal anyway.
* clients should NOT login. Since the license counts "login users" according to the website, I assume I cannot allow "normal users" to log in in the custom client, since it would create a login user, wouldn't it?

We must have a way to block this, otherwise our InfoSec will never allow Rustdesk to go live in production in our network, if nobody can explain how this happens. I'd expect the custom clients to exclusively work with our server, since it distributes its own public key. Nobody else should have it.

**Reply by Nawaki89** [reply]
Regarding "clients should NOT login"  this is correct if you enable the login feature on the server its only needed that the controlling side the one that is connecting to remote control a PC is logged in.   not the client its connecting to. 

this is how I do, we are 5 ppl remote controlling and is logged in targeting 350 PCs that is not logged in since they are only endpoints that need assistance not doing any remote connecting.

**Comment by OtterKring** [comment]
I am still trying to make sense out those ghost devices appearing on our server out of nowhere.

Since our self-hosted server is using a private/public key pair which is rolled out to the clients to connect, I cannot think of another possibility of someone getting hold of our custom client installation package after it is built. The builds happen on RustDesk's infrastructure, so our data is offloaded to their servers for the compilation and we can download the completed package from there.
The packages might be removed from their servers after a day, but in the meantime, if someone gets access to the download section, they can access the custom client and connect to our server.

This is NOT good. If there is no fix for this, InfoSec will not approve the software (we are still in PoC).

**Reply by dinger1986** [reply]
its AV doing cloud heuristics https://www.reddit.com/r/atera/comments/ckay22/random_computers_showing_up_in_device_list/

Happens with almost every remote system
